it/src/main/java/com/google/cloud/teleport/it/pubsublite/PubsubLiteResourceManager.java

```diff
@@ -0,0 +1,216 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.pubsublite;
+
+import com.google.cloud.pubsublite.AdminClient;
+import com.google.cloud.pubsublite.AdminClientSettings;
+import com.google.cloud.pubsublite.CloudRegion;
+import com.google.cloud.pubsublite.ProjectId;
+import com.google.cloud.pubsublite.ReservationName;
+import com.google.cloud.pubsublite.ReservationPath;
+import com.google.cloud.pubsublite.SubscriptionName;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.TopicName;
+import com.google.cloud.pubsublite.TopicPath;
+import com.google.cloud.pubsublite.proto.Reservation;
+import com.google.cloud.pubsublite.proto.Subscription;
+import com.google.cloud.pubsublite.proto.Topic;
+import com.google.protobuf.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/** Interface for managing Pub/Sub resources in integration tests. */
+public interface PubsubLiteResourceManager {
+
+  /**
+   * Create a PubsubLite reservation with {@code capacity} number of capacity units, which represent
+   * {@code 0.25} MiBps on a regional reservation.
+   *
+   * @param reservationName
+   * @param cloudRegion
+   * @param capacity
+   * @return
+   */
+  ReservationPath createReservation(
+      String reservationName, String cloudRegion, String projectId, Long capacity);
+
+  /**
+   * Creates a topic with the given name on Pub/Sub.
+   *
+   * <p>https://cloud.google.com/pubsub/lite/docs/reservations
+   *
+   * @param topicName Topic name to create. The underlying implementation may not use the topic name
+   *     directly, and can add a prefix or a suffix to identify specific executions.
+   * @return The instance of the TopicName that was just created.
+   */
+  TopicName createTopic(String topicName, ReservationPath reservationPath);
+
+  /**
+   * Creates a subscription at the specific topic, with a given name.
+   *
+   * @param topicName Topic Name reference to add the subscription.
+   * @param subscriptionName Name of the subscription to use. Note that the underlying
+   *     implementation may not use the subscription name literally, and can use a prefix or a
+   *     suffix to identify specific executions.
+   * @return The instance of the SubscriptionName that was just created.
+   */
+  SubscriptionName createSubscription(
+      ReservationPath reservationPath, TopicName topicName, String subscriptionName);
+
+  /** Delete any topics or subscriptions created by this manager. */
+  void cleanupAll();
+
+  /** Default exception type for {@link PubsubLiteResourceManager}. */
+  class PubsubliteResourceManagerException extends Exception {}
+
+  /** Default implementation for {@link PubsubLiteResourceManager}. */
+  class DefaultPubsubliteResourceManager implements PubsubLiteResourceManager {
+
```

Comment:
nit: Can you move both of those to separate files, similar to other resource managers?

Reply:
done